app/gui/view/graph-editor/src/component/visualization/foreign/java_script/visualization.js

```diff
@@ -29,56 +29,25 @@ export class Visualization {
      */
     __emitPreprocessorChange__() {
         this.__api__.emit_preprocessor_change(
-            this.__preprocessorCode__,
-            this.__preprocessorModule__
+            this.__preprocessorModule__,
+            this.__preprocessorMethod__
         )
     }
 
     /**
-     * Get the current preprocessor code. See {@link setPreprocessorCode} for
-     * more information about purpose of setting preprocessor code.
+     * Get the current preprocessor method. See {@link setPreprocessor} for
+     * more information about purpose of setting the preprocessor.
      *
-     * @returns {string} Preprocessor code or `null` if no code was set.
+     * @returns {string} Preprocessor method or `null` if no method was set.
      */
-    getPreprocessorCode() {
-        return this.__preprocessorCode__
-    }
-
-    /**
-     * Set new preprocessor code.
-     *
-     * When visualization is attached to a node, each time a new value is produced from node,
-     * the preprocessor shall be invoked with it. Result such call shall be serialized and
-     * transported to visualization by invoking onDataReceived(data) method.
```

Comment:
I miss such detailed documentation in the new version.

We have a special documentation page for contributors wanting to create new visualizations, and these docs AFAIK refers to this file for specific documentation of JS methods they can use inside their visualizations.

Reply:
I updated the doc with more details and examples. Should be better now